core/src/main/java/org/infinispan/manager/EmbeddedCacheManagerAdmin.java

```diff
@@ -68,6 +68,37 @@
     */
    <K, V> Cache<K, V> getOrCreateCache(String name, Configuration configuration);
 
+   /**
+    * Creates a template that is replicated across the cluster using the specified configuration.
+    * The template will survive topology changes, e.g. when a new node joins the cluster,
+    * it will automatically be created there. This method will wait for the template to be created on all nodes before
+    * returning.
+    *
+    * @param name the name of the template
+    * @param configuration the configuration to use. It must be a clustered configuration (e.g. distributed)
+    * @throws org.infinispan.commons.CacheConfigurationException if a template with the same name already exists
+    */
+   void createTemplate(String name, Configuration configuration);
+
+   /**
+    * Retrieves an existing template or creates one across the cluster using the specified configuration.
+    * The template will survive topology changes, e.g. when a new node joins the cluster,
+    * it will automatically be created there. This method will wait for the template to be created on all nodes before
+    * returning.
+    *
+    * @param name the name of the template
+    * @param configuration the configuration to use. It must be a clustered configuration (e.g. distributed)
+    * @return the template configuration
+    */
+   Configuration getOrCreateTemplate(String name, Configuration configuration);
+
+   /**
+    * Removes a template from the cache container. Any persisted data will be cleared.
```

Comment:
Shouldn't it say "across the cluster" as well? E.g.

`Removes a template across the cluster. If the template is persistent, it will also be removed from persisted state`.